schemas/expconf/v0/s3.json

```diff
@@ -42,6 +42,14 @@
             ],
             "default": null
         },
+        "prefix": {
+            "type": [
+                "string",
+                "null"
+            ],
+            "pattern": "^([^/].*[^/]|[^/])$",
```

Comment:
The only pattern I would bother enforcing is that `..` doesn't appear.  But even that, if the aws http server rejects it, might be unnecessary.

If you do choose to have a pattern here, please add testcases to ensure it's working.

Reply:
Hrm... I guess we could allow a user to put anything here.

I guess we need to define what is supposed to happen given different inputs:
```
The following would all yield: /bucket/a/storage_id/
/a/
a/
/a
a
All of these would yield: "/bucket/storage_id"
/
///
""
None

All these would fail on config parse:
..
/..
/abc/../foo
```
^ Does this behavior seem correct? We are trying to make these some semblance of a unix path, even though I think AWS supports any chars in the key name.